plugin.rb

```diff
@@ -1,13 +1,94 @@
 # frozen_string_literal: true
 
-# name: discourse-plugin-name
-# about: TODO
-# version: 0.0.1
+# name: discourse-cdn-experiment
+# about: Allows A/B testing multiple CDN domains
+# version: 1.0
 # authors: Discourse
-# url: TODO
-# required_version: 2.7.0
+# url: https://github.com/discourse/discourse-cdn-experiment
 
-enabled_site_setting :plugin_name_enabled
+enabled_site_setting :cdn_experiment_enabled
+
+module ::CdnExperiment
+  ENV_KEY = "discourse-cdn-experiment-cdn-index"
+
+  def self.perform_gsub(value, env)
+    if value.is_a? String
+      was_html_safe = value.html_safe?
+      value = value.gsub(GlobalSetting.cdn_url, app_cdn_url(env))
+      value = value.gsub(GlobalSetting.s3_cdn_url, s3_cdn_url(env))
+      value = value.html_safe if was_html_safe
+      value
+    elsif value.is_a? Hash
+      value.transform_values { |hash_value| perform_gsub(hash_value, env) }
+    else
+      value
+    end
+  end
+
+  def self.s3_cdn_url(env)
+    s3_cdn_urls[current_cdn_index(env)]
+  end
+
+  def self.app_cdn_url(env)
+    app_cdn_urls[current_cdn_index(env)]
```

Comment:
We need to use the same index for both the `app_cdn_urls` and the `s3_cdn_urls`. If we used sample, it might return the first app_cdn, but the second s3_cdn

Reply:
Ah, I see. Thanks!